src/report.ts

```diff
@@ -4,9 +4,17 @@ import * as models from 'powerbi-models';
 import * as wpmp from 'window-post-message-proxy';
 import * as hpm from 'http-post-message';
 import * as utils from './util';
+import { IFilterable } from './ifilterable';
+import { IPage, Page } from './page';
```

Comment:
Just pointing out the circular reference here.  I always try to avoid them if possible, but in this case it was required.  

The page class needs reference to the report that it belongs to and the report needs to be able to construct pages and return array of pages.  If there is better way to handle this that avoids circular reference i'd be interested in feedback.